src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Telemetry/ITelemetryReporter.cs

```diff
@@ -11,9 +11,9 @@ internal interface ITelemetryReporter
     TelemetryScope BeginBlock(string name, Severity severity, Property property);
     TelemetryScope BeginBlock(string name, Severity severity, Property property1, Property property2);
     TelemetryScope BeginBlock(string name, Severity severity, Property property1, Property property2, Property property3);
-    TelemetryScope BeginBlock(string name, Severity severity, params Property[] properties);
```

Comment:
❗ The purpose of the overloads above is to avoid allocating a params array in cases where there are zero to three properties. This change will cause any calls that now pass a `TimeSpan` and zero to three properties to create an extra array: one for the params array, and one for the final telemetry array of length + . I recommend adding `TimeSpan` to each of these overloads and then adding extension methods on `ITelemetryReporter` that don't have the `TimeSpan` parameter and just pass `TimeSpan.Zero`.

Also, consider turning the final overload into `params ReadOnlySpan<Property>` to match `ReportEvent` below.

Reply:
I think `minTimeToReport` seems weird on BeginBlock. Maybe it would be better to add something to `TelemetryScope` for this?

Reply:
You're right. It does feel a bit strange on a method called `BeginBlock`. However, I think it would be awkward ergonomically to have some property to set on `TelemetryScope`. There are two reasons for this:

1. Nothing actually uses a `TelemetryScope` today. It's always captured in a using statement.
2. There's a race condition between the `TelemetryScope` being created and the value being set. So, passing it as a parameter so that it can be included in the `TelemetryScope` is probably the right thing.

Of course, the truth is that `BeginBlock` is kind of a weird name because since it produces a `TelemetryScope`. Also, `TelemetryScope` is currently a class. Should it become a struct with `[NonCopyable]` to avoid the extra allocation?

Reply:
> Nothing actually uses a TelemetryScope today. It's always captured in a using statement.

But they could. Scopes probably should have a cancel anyways... 

I like the idea of them being a `[NonCopyable]` struct. I still find it odd to add a timespan to every `BeginBlock` when only `TrackLspRequest` is using it. It seems like we're taking an implementation of one part of the API surface and expanding it. 

In the end it's not a huge deal though, and with the added extension methods it won't cause any real pain

Reply:
> But they could. Scopes probably should have a cancel anyways... 

Maybe? At the moment that's kind of YAGNI.